File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        int len = arr.size();
        int ans = 0;
        for(int i=0; i<len; i++){
            int a_xor = arr[i];
            for(int j=i+1; j<len; j++){
                a_xor ^= arr[j];
                if(a_xor == 0){
                    ans += (j-i);
                }
            }
        }
        return ans;
    }
};

File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var countTriplets = function(arr) {
    var ans = 0;
    for(var i=0; i<arr.length; i++){
        var xor_a = arr[i];
        for(var j=i+1; j<arr.length; j++){
            xor_a ^= arr[j]
            if(xor_a === 0){
                ans += (j-i);
            }
        }
    }
    return ans;
};